yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -303,7 +303,9 @@ describe('AVM simulator: transpiled Noir contracts', () => {
 
       // Note hash existence check should be in trace
       const trace = context.persistableState.flush();
-      expect(trace.noteHashChecks).toEqual([expect.objectContaining({ noteHash, leafIndex, exists: false })]);
+      expect(trace.noteHashChecks).toEqual(
+        expect.arrayContaining([expect.objectContaining({ noteHash, leafIndex, exists: false })]),
```

Comment:
IIUC `arrayContaining` matches a subset. Do we want this? https://jestjs.io/docs/expect#expectarraycontainingarray